test/adapter/json_api/links_test.rb

```diff
@@ -0,0 +1,37 @@
+require 'test_helper'
+
+LinkPost = Class.new(Model)
+class LinkPostSerializer < ActiveModel::Serializer
+  belongs_to :link_author
+end
+
+LinkAuthor = Class.new(Model)
+class LinkAuthorSerializer < ActiveModel::Serializer
+  links do
+    link :self, href: "//link_author/#{object.id}"
+  end
+end
+
+module ActiveModel
+  class Serializer
+    module Adapter
+      class JsonApi
+        class LinksTest < Minitest::Test
+          def setup
+            @author = LinkAuthor.new(id: 1337)
+          end
+
+          def test_resource_links
+            hash = ActiveModel::SerializableResource.new(@author, adapter: :json_api).serializable_hash
```

Comment:
Same answer as in my other PR, I'm all up for this helper, but I do not want to redefine it here. We should move it to `test_helper.rb`.